File: app/src/main/java/com/imperium/academio/ui/model/MaterialItemRvModel.java
package com.imperium.academio.ui.model;

import androidx.annotation.Nullable;

public class MaterialItemRvModel {
    private final int icon;
    private final String title;
    private final String subtitle;
    private final String key;

    public MaterialItemRvModel(int icon, String title, String subtitle, String key) {
        this.icon = icon;
        this.title = title;
        this.subtitle = subtitle;
        this.key = key;
    }

    public int getIcon() {
        return icon;
    }

    public String getTitle() {
        return title;
    }

    public String getSubtitle() {
        return subtitle;
    }

    @Override
    public boolean equals(@Nullable Object obj) {
        if (!(obj instanceof MaterialItemRvModel))
            return false;
        MaterialItemRvModel model = (MaterialItemRvModel) obj;
        return model.key.equals(key) && model.title.equals(title);
    }
}

File: app/src/main/java/com/imperium/academio/ui/model/MaterialItemRvModel.kt
package com.imperium.academio.ui.model

class MaterialItemRvModel(val icon: Int?, val title: String?, val subtitle: String?, private val key: String?) {
    override fun equals(other: Any?): Boolean {
        if (other !is MaterialItemRvModel) return false
        return other.key == key && other.title == title
    }

    override fun hashCode(): Int {
        var result = title.hashCode()
        result = 31 * result + key.hashCode()
        return result
    }
}